client/src/components/App/Content/InvestigationForm/TabManagement/TabManagement.tsx

```diff
@@ -3,90 +3,108 @@ import { Tabs, Tab, Card, createStyles, withStyles } from '@material-ui/core';
+            wrapper: {
+                flexDirection: "row-reverse",
+            }
         }),
     )(Tab);
 
-  const handleTabChange = (event: React.ChangeEvent<{}>, selectedTab: number) => {
-    setCurrentTab({
-        id: selectedTab,
-        name: tabs[selectedTab].name, 
-        displayComponent: tabs[selectedTab].displayComponent,
-        isDisabled: false,
-    });
-  };
+    const investigationId = useSelector<StoreStateType, number>((state) => state.investigation.epidemiologyNumber);
+    const formsValidations = useSelector<StoreStateType, any>((state) => state.formsValidations[investigationId]);
 
     return (
         <Card className={classes.card}>
                 <Tabs
-                    value={currentTab.id}
+                    value={currentTab}
                     indicatorColor='primary'
                     textColor='primary'
-                    onChange={(event, selectedTab) => !shouldDisableChangeTab && handleTabChange(event, selectedTab)}
                 >
                     {
                         tabs.map((tab) => {
-                            return <StyledTab onClick={onTabClicked} key={tab.id} label={tab.name} disabled={tab.isDisabled}/>
+                            if (formsValidations !== undefined && !formsValidations[tab.id] && formsValidations[tab.id] !== null) {
+                                return <StyledTab 
+                                // @ts-ignore
+                                type="submit"
+                                form={`form-${currentTab}`}
+                                onClick={() => {setNextTab(tab.id)}}
+                                key={tab.id}
+                                label={tab.name}
+                                icon={<ErrorOutlineIcon/>}
+                                className={classes.errorIcon}
+                            />} else {
+                                    return <StyledTab 
+                                        // @ts-ignore
+                                        type="submit"
+                                        form={`form-${currentTab}`}
+                                        onClick={() => {setNextTab(tab.id)}}
+                                        key={tab.id}
+                                        label={tab.name}
+                                    />}
                         })
                     }
                 </Tabs>
-            {
-                <div key={currentTab.id} className={classes.displayedTab}>
-                    {currentTab.displayComponent }
+            
+                <div className={classes.displayedTab}>
+                {currentTab == 0 && <PersonalInfoTab id={0} onSubmit={moveToNextTab}/>}
+                {currentTab == 1 && <ClinicalDetails id={1} onSubmit={moveToNextTab}/>}
+                {currentTab == 2 && <ExposuresAndFlights id={2} onSubmit={moveToNextTab}/>}
+                {currentTab == 3 && <InteractionsTab id={3} onSubmit={moveToNextTab}/>}
```

Comment:
I don't know why but it's not pretty to me design wise,  why removing the display component?

Reply:
breaks open close principal

Reply:
fixed, i put back the display component in the array